nextjs/packages/next/server/config-shared.ts

```diff
@@ -171,6 +171,9 @@ export type NextConfig = { [key: string]: any } & {
     pageDataCollectionTimeout?: number
     isrMemoryCacheSize?: number
     concurrentFeatures?: boolean
+  }  
+  template?:{
+    typeToComponentMap: any
```

Comment:
Could it be something like `typeToComponentMap: [key in PossibleTypes]: string`?

Reply:
If `PossibleTypes` (for generator, I only found `zodPossibleTypes`) doesn't exist, we could add one — what do you think?